### PR TITLE
2 hz multilayerWU OrdMLWU OrdMxBUD OrdMlWD

### LINKED ISSUE #2: Multilayer Graphs
Create:
- [ ] _MultilayerWD.gen.m (Blanca)
- [x] #12 (Hang)
- [x] _MultilayerBD.gen.m (Blanca)
- [x] #19

Create:
- [x] _OrdMlWD.gen.m (Hang)
- [x] _OrdMlWU.gen.m (Hang)
- [ ] _OrdMlBD.gen.m (Blanca)
- [ ] _OrdMlBU.gen.m (jws)

Create
- [x] _OrdMxBUD.gen.m (Hang)
- [x] _OrdMxBUT.gen.m (Blanca)
- [x] #11 (jws)
- [x] _MultilayerBUT.gen.m (Blanca)
- [ ] _OrdMlBUD.gen.m (jws)
- [ ] _OrdMlBUT.gen.m (jws)

### LINKED ISSUE #16: Graph Type check (Ordinal Multilayer Graph)
In the graph type check,

The code should raise an error when the number of columns per submatrix in the same column is not the same, but in the code, it is the other way around that the error is raised when the number of columns per submatrix in the same column is the same.

Change '==' to '~=' would fix.

### PR DESCRIPTION
**Pull request by hanzh346**
OrdMxBUD.gen.m created,OrdMlWU.gen.m created, MultilayerWU.gen.m created, OrdMlWD.gen.m created unitest passed, GUI checked

### CODE REVIEW

**Review by bzufiriaki — APPROVED** [review]

File: braph2genesis/graphs/_MultilayerWU.gen.m [hunk 1]
```diff
@@ -0,0 +1,256 @@
+%% ¡header!
+MultilayerWU < Graph (g, multilayer weighted undirected graph) is a multilayer weighted undirected graph.
+
+%%% ¡description!
+In a multilayer weighted undirected (WU) graph, layers could have different number 
+ of nodes with within-layer weighted undirected edges, associated with a real 
+ number between 0 and 1 and indicating the strength of the connection.
+The connectivity matrices are symmetric.
+There are connections between layers connecting the corresponding nodes.
```
Comment (annacanal) [inline braph2genesis/graphs/_MultilayerWU.gen.m]: I would substitute these two sentences by:
**The connectivity matrices are symmetric (within layer).
All node connections are allowed between layers.**

File: braph2genesis/graphs/_MultilayerWU.gen.m [hunk 2]
```diff
@@ -0,0 +1,256 @@
...
+%%% ¡prop!
+
+NAME (constant, string) is the name of the multilayer weighted undirected graph.
+%%%% ¡default!
+'MultilayerWU'
+
+%%% ¡prop!
+DESCRIPTION (constant, string) is the description of the multilayer weighted undirected graph.
+%%%% ¡default!
+'In a multilayer weighted undirected (WU) graph, layers could have different number of nodes with within-layer weighted undirected edges, associated with a real number between 0 and 1 and indicating the strength of the connection. The connectivity matrices are symmetric. There are connections between layers connecting the corresponding nodes.'
```
Comment (annacanal) [inline braph2genesis/graphs/_MultilayerWU.gen.m]: update here the description based on the previous comment

File: braph2genesis/graphs/_OrdMlWU.gen.m [hunk 3]
```diff
@@ -0,0 +1,259 @@
+%% ¡header!
+OrdMlWU < Graph (g, ordinal multilayer weighted undirected graph) is an ordinal multilayer weighted undirected graph.
+
+%%% ¡description!
+In an ordinal multilayer weighted undirected (WU) graph, layers could have different number 
+ of nodes with within-layer weighted undirected edges, associated with a real 
+ number between 0 and 1 and indicating the strength of the connection.
+The connectivity matrices are symmetric.
+There are connections between layers connecting the corresponding nodes.
+The layers are connected in an ordinal fashion.
```
Comment (annacanal) [inline braph2genesis/graphs/_OrdMlWU.gen.m]: I would substitute these three sentences by:
**The connectivity matrices are symmetric (within layer).
All node connections are allowed between layers.
The layers are connected in an ordinal fashion, i.e., only consecutive layers are connected.**

File: braph2genesis/graphs/_OrdMlWU.gen.m [hunk 4]
```diff
@@ -0,0 +1,259 @@
...
+
+NAME (constant, string) is the name of the ordinal multilayer weighted undirected graph.
+%%%% ¡default!
+'OrdMlWU'
+
+%%% ¡prop!
+DESCRIPTION (constant, string) is the description of the ordinal multilayer weighted undirected graph.
+%%%% ¡default!
+'In an ordinal multilayer weighted undirected (WU) graph, layers could have different number of nodes with within-layer weighted undirected edges, associated with a real number between 0 and 1 and indicating the strength of the connection. The connectivity matrices are symmetric. There are connections between layers connecting the corresponding nodes.The layers are connected in an ordinal fashion.'
+
```
Comment (annacanal) [inline braph2genesis/graphs/_OrdMlWU.gen.m]: update the description

File: braph2genesis/graphs/_OrdMlWU.gen.m [hunk 5]
```diff
@@ -0,0 +1,259 @@
...
+%%%% ¡calculate!
+if isempty(varargin)
+    layernumber = 1;
+else
+    layernumber = varargin{1};
+end
+value = Graph.NONNEGATIVE * ones(layernumber);
+
+%%% ¡prop!
+A (result, cell) is the cell containing the multilayer weighted adjacency matrices of the multilayer weighted undirected graph.
```
Comment (annacanal) [inline braph2genesis/graphs/_OrdMlWU.gen.m]: it contains more than adjacency matrices. So either you say that it contains the **multilayer weighted supra-adjacency matrix** or you say that it contains the within-layer weighted adjacency matrices of the multilayer weighted undirected graph and the ordinal connections between layers.

File: braph2genesis/graphs/_MultilayerWU.gen.m [hunk 6]
```diff
@@ -0,0 +1,256 @@
...
+%%%% ¡calculate!
+if isempty(varargin)
+    layernumber = 1;
+else
+    layernumber = varargin{1};
+end
+value = Graph.NONNEGATIVE * ones(layernumber);
+
+%%% ¡prop!
+A (result, cell) is the cell containing the multilayer weighted adjacency matrices of the multilayer weighted undirected graph.
```
Comment (annacanal) [inline braph2genesis/graphs/_MultilayerWU.gen.m]: it contains more than adjacency matrices. So either you say that it contains the multilayer weighted supra-adjacency matrix or you say that it contains the within-layer weighted adjacency matrices of the multilayer weighted undirected graph and the connections between layers.

File: braph2genesis/graphs/_OrdMxBUD.gen.m [hunk 7]
```diff
@@ -0,0 +1,203 @@
+%% ¡header!
+OrdMxBUD < OrdMxWU (g, binary undirected multiplex with fixed densities) is a ordinal multiplex binary undirected with fixed densities.
+
+%%% ¡description!
+In a ordinal binary undirected multiplex with fixed densities (BUD), the layers are 
+ those of binary undirected (BU) multiplex graphs derived from the same 
+ weighted supra-connectivity matrices binarized at different densities.
+The supra-connectivity matrix has a number of partitions equal to the number of densities.
+The layers are connected in an ordinal fashion, i.e., only consecutive layers are connected.
```
Comment (annacanal) [inline braph2genesis/graphs/_OrdMxBUD.gen.m]: There is just **one supra-adjacency matrix** in the weighted multiplex/multilayer, if you talk in plural you mean the within-layer connectivity matrix. Also talk about the supra-adjacency matrix, not the supra connectivity matrix.

Comment (annacanal) [reply braph2genesis/graphs/_OrdMxBUD.gen.m]: Also: In an ordinal... ; this needs to be rewritten.

File: braph2genesis/graphs/_OrdMxBUD.gen.m [hunk 8]
```diff
@@ -0,0 +1,203 @@
...
+%%% ¡prop!
+NAME (constant, string) is the name of the ordinal binary undirected multiplex with fixed densities.
+%%%% ¡default!
+'OrdMxBUD'
+
+%%% ¡prop!
+DESCRIPTION (constant, string) is the description of the ordinal binary undirected multiplex with fixed densities.
+%%%% ¡default!
+'In a ordinal binary undirected multiplex with fixed densities (BUD), the layers are those of binary undirected (BU) multiplex graphs derived from the same weighted supra-connectivity matrices binarized at different densities. The supra-connectivity matrix has a number of partitions equal to the number of densities.The layers are connected in an ordinal fashion, where just consecutive layers are connected'
+
```
Comment (annacanal) [inline braph2genesis/graphs/_OrdMxBUD.gen.m]: update

File: braph2genesis/graphs/_OrdMxBUD.gen.m [hunk 9]
```diff
@@ -0,0 +1,203 @@
...
+%%%% ¡calculate!
+if isempty(varargin)
+    layernumber = 1;
+else
+    layernumber = varargin{1};
+end
+value = Graph.NONNEGATIVE * ones(layernumber);
+
+%%% ¡prop!
+A (result, cell) is the cell array containing the multiplex binary adjacency matrices of the binary undirected multiplex.
```
Comment (annacanal) [inline braph2genesis/graphs/_OrdMxBUD.gen.m]: is the cell array containing the multiplex binary **supra**-adjacency matrices of the binary undirected multiplex

**Review by annacanal — CHANGES_REQUESTED** [review]
Descriptions need to be modified in the 3 graphs implemented in this pull request, see my recommendations in each comment

**Review by JiaweiSun0512 — APPROVED** [review]

File: temp.m
Comment (iok-ui) [inline temp.m:1]: This file should not be updated, according to last discussion

File: braph2genesis/measures/_Distance.gen.m [hunk 10]
```diff
@@ -945,3 +945,204 @@
...
+    0       0   0   0    0  0 
+    ];
+B22 = [
+    0       .1  .2  .25  0;
+    .125    0   0   0    0;
+    .2      .5  0   .25  0;
+    .125    10  0   0    0;
+    0       0   0   0    0;
+    ];
+B12 = rand(size(B11,1),size(B22,2));
```
Comment (iok-ui) [inline braph2genesis/measures/_Distance.gen.m:1133]: Leave a space after comman, i.e. 
"B12 = rand(size(B11, 1), size(B22, 2));"
And distribute the changes across the script and keep the style consistent

**Review by iok-ui — CHANGES_REQUESTED** [review]

File: braph2genesis/graphs/_MultilayerWU.gen.m [hunk 11]
```diff
@@ -90,7 +90,7 @@ In a multilayer weighted undirected (WU) graph, layers could have different numb
 value = Graph.NONNEGATIVE * ones(layernumber);
 
 %%% ¡prop!
-A (result, cell) is the cell containing the multilayer weighted adjacency matrices of the multilayer weighted undirected graph.
+A (result, cell) is the cell containing the multilayer weighted supra-adjacency matrices of the multilayer weighted undirected graph.
```
Comment (annacanal) [inline braph2genesis/graphs/_MultilayerWU.gen.m]: A (result, cell) is the cell containing the weighted supra-adjacency matrix of the multilayer weighted undirected graph.

File: braph2genesis/graphs/_OrdMxBUD.gen.m [hunk 12]
```diff
@@ -88,7 +88,7 @@ those of binary undirected (BU) multiplex graphs derived from the same
 value = Graph.NONNEGATIVE * ones(layernumber);
 
 %%% ¡prop!
-A (result, cell) is the cell array containing the multiplex binary adjacency matrices of the binary undirected multiplex. 
+A (result, cell) is the cell array containing the multiplex binary supra-adjacency matrices of the binary undirected multiplex.
```
Comment (annacanal) [inline braph2genesis/graphs/_OrdMxBUD.gen.m:91]: A (result, cell) is the cell array containing the binary supra-adjacency matrix of the binary undirected multiplex.

**Review by iok-ui — CHANGES_REQUESTED** [review]
Overall it is good. Two things before merging:

1.  there should not be any changes for the temp.m file
2. there are conflicts in the measures (Degree, DegreeAv, Distance). From first glance, it seems due to OrdMxBUT implemnted in the develop. This changes of OrdMxBUT needs to be coped with into this PR

**Review by annacanal — APPROVED** [review]

**Review by iok-ui — APPROVED** [review]

**Review by egolol — APPROVED** [review]

**Pull request merged**